test/extended/router/config_manager.go

```diff
@@ -52,8 +57,439 @@ var _ = g.Describe("[sig-network][Feature:Router][apigroup:route.openshift.io][a
+									ValueFrom: &corev1.EnvVarSource{
+										FieldRef: &corev1.ObjectFieldSelector{
+											FieldPath: "metadata.namespace",
+										},
+									},
+								},
+							},
+							Args: []string{
+								"--namespace=$(POD_NAMESPACE)",
+								"-v=4",
+								"--haproxy-config-manager=true",
+								"--blueprint-route-labels=select=hapcm-blueprint",
+								"--labels=select=haproxy-cfgmgr",
+								"--stats-password=password",
+								"--stats-port=1936",
+								"--stats-user=admin",
+							},
+							Ports: []corev1.ContainerPort{
+								{
+									ContainerPort: 80,
+								},
+								{
+									ContainerPort: 443,
+								},
+								{
+									ContainerPort: 1936,
+									Name:          "stats",
+									Protocol:      corev1.ProtocolTCP,
+								},
+							},
+						},
+					},
+				},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "insecure-endpoint",
+					Labels: map[string]string{
+						"test":      "haproxy-cfgmgr",
+						"endpoints": "insecure-endpoint",
+					},
+				},
+				Spec: corev1.PodSpec{
+					TerminationGracePeriodSeconds: utilpointer.Int64(1),
+					Containers: []corev1.Container{
+						{
+							Name:  "test",
+							Image: image.LocationFor("registry.k8s.io/e2e-test-images/agnhost:2.40"),
+							Args:  []string{"netexec"},
+							Ports: []corev1.ContainerPort{
+								{
+									ContainerPort: 8080,
+									Name:          "http",
+								},
+								{
+									ContainerPort: 100,
+									Protocol:      corev1.ProtocolUDP,
+								},
+							},
+						},
+					},
+				},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "secure-endpoint",
+					Labels: map[string]string{
+						"app": "secure-endpoint",
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:    "server",
+							Image:   "k8s.gcr.io/e2e-test-images/nginx:1.15-2",
+							Command: []string{"/usr/sbin/nginx"},
+							Args:    []string{"-c", "/etc/nginx/nginx.conf"},
+							Ports: []corev1.ContainerPort{
+								{
+									ContainerPort: 8443,
+									Protocol:      corev1.ProtocolTCP,
+								},
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "cert",
+									MountPath: "/etc/serving-cert",
+								},
+								{
+									Name:      "conf",
+									MountPath: "/etc/serving-cert",
+								},
+								{
+									Name:      "tmp",
+									MountPath: "/var/cache/nginx",
+								},
+								{
+									Name:      "tmp2",
+									MountPath: "/var/run",
+								},
+							},
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: "conf",
+							VolumeSource: corev1.VolumeSource{
+								ConfigMap: &corev1.ConfigMapVolumeSource{
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: "serving-cert",
+									},
+								},
+								Secret: &corev1.SecretVolumeSource{
+									SecretName: "service-cert-grpc-interop",
+								},
```

Comment:
I think this is copypasta.  The volume only needs (and only can have) one volume source.

Reply:
Removed